# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLBlitHelper.cpp
# Commit: 20a5670ea914
# Full Hash: 20a5670ea9141e93bcdeca586122a82ba1ce8d49
# Author: stransky <stransky@redhat.com>
# Date: 2025-01-08 04:42:02
# Regressor Bug: 1940037
# File Overlap Count: 1
# Description:
#   Bug 1940037 [Wayland] Use SURFACE_YUV/SURFACE_RGBA dmabuf surface type only r=emilio
#   
#   We need to add exact FOURCC dmabuf surface format with correct modifers so use RGBA/YUV surface type only.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D233240
# ==============================================================================

diff -r 38d1226c5470 -r 20a5670ea914 gfx/gl/GLBlitHelper.cpp
--- a/gfx/gl/GLBlitHelper.cpp	Tue Jan 07 11:50:02 2025 +0000
+++ b/gfx/gl/GLBlitHelper.cpp	Tue Jan 07 11:58:50 2025 +0000
@@ -1566,18 +1566,20 @@
     case DMABufSurface::SURFACE_RGBA:
       fragSample = kFragSample_OnePlane;
       break;
-    case DMABufSurface::SURFACE_NV12:
-      fragSample = kFragSample_TwoPlane;
-      pYuvArgs = &yuvArgs;
-      fragConvert = kFragConvert_ColorMatrix;
-      break;
-    case DMABufSurface::SURFACE_YUV420:
-      fragSample = kFragSample_ThreePlane;
+    case DMABufSurface::SURFACE_YUV:
+      if (surface->GetTextureCount() == 2) {
+        fragSample = kFragSample_TwoPlane;
+      } else if (surface->GetTextureCount() == 3) {
+        fragSample = kFragSample_ThreePlane;
+      } else {
+        gfxCriticalError() << "Unexpected planes count: "
+                           << surface->GetTextureCount();
+        return false;
+      }
       pYuvArgs = &yuvArgs;
       fragConvert = kFragConvert_ColorMatrix;
       break;
     default:
-      gfxCriticalError() << "Unexpected pixel format: " << pixelFormat;
       return false;
   }
 